Task 290 import export source code

deadlexus opened the pull request:
#290

Commit:
MSI: issue #290
- Adapted SourceItem Import/Export to use Source Code instead of SourceId;
- Fixed issue when in Stock Sources export process all fields ware excluded;
- Fixed type of return value for SourceItem->getSourceId method to int(was string);
- Adapted integrational test to the new Export/Import system(with source code).

Commit:
Merge remote-tracking branch 'origin/task-288-add-source-code' into task-290-import-export-source-code

Conflicts:
	app/code/Magento/InventoryApi/Test/_files/source.php
	app/code/Magento/InventoryApi/Test/_files/sources.php

Commit:
MSI: issue #290
- Fixed integration test.

Commit:
MSI: issue #290
- Fixed DependencyTest error.

maghamed commented on app/code/Magento/Inventory/Model/SourceItem.php:
```diff
@@ -48,7 +48,7 @@ public function setSku($sku)
     public function getSourceId()
     {
-        return $this->getData(self::SOURCE_ID);
+        return (int)$this->getData(self::SOURCE_ID);
```
It's not correct to cast to `int` here, 
because when the factory creates SourceId there is no guarantee that factory will set something to SourceId

that's why PHP Doc says (` @return int|null `)

alexey-yakimovich committed:
MSI: issue #290
- Changed SourceItem->getSourceId method.

alexey-yakimovich committed:
MSI: issue #290
- MEQP2 fixes;

maghamed committed:
Merge branch 'develop' into task-290-import-export-source-code

Commit:
MSI: issue #290
- modified Source Item database structure(source_id changed to source_code);
- SourceId was replaced by SourceCode in MSI Extensions;

alexey-yakimovich committed:
MSI: issue #290
- source_id field usings ware replaced by source_code field - part;

alexey-yakimovich committed:
MSI: issue #290
- fixed foreign key issue with source_item_configuration setup schema;
- tests fixes - part;

alexey-yakimovich committed:
MSI: issue #290
- Fixed setup schema script for creatind default source;
- Tests fexes - part;

alexey-yakimovich committed:
MSI: issue #290
- fixed install schema/data issues;
- fixed issues on Source grid;
- identity field returned to source_id on Source mode;

alexey-yakimovich committed:
Merge remote-tracking branch 'origin/develop' into task-290-import-export-source-code

Conflicts:
	app/code/Magento/Inventory/Controller/Adminhtml/Stock/StockSourceLinkProcessor.php

alexey-yakimovich committed:
MSI: issue #290
- Fixed issue with with incorrect query on product edit page;

alexey-yakimovich committed:
MSI: issue #290
- Fixed issue with source item UIcomponent on product edit page

alexey-yakimovich committed:
Merge remote-tracking branch 'origin/develop' into task-290-import-export-source-code

Conflicts:
	app/code/Magento/InventoryApi/Test/_files/products.php

alexey-yakimovich committed:
MSI: issue #290
- Fixed issued with UI Components on stock edit page;

alexey-yakimovich committed:
MSI: issue #290
- Integration test fixes - part;

alexey-yakimovich committed:
MSI: issue #290
- Integration tests fixes - part;

Commit:
MSI: issue #290
- Test fixes -part;

maghamed committed:
Merge branch 'develop' into task-290-import-export-source-code

maghamed committed:
Adapt SourceCodes with PreventAssignSourcesToDefaultStockPlugin logic

maghamed committed:
Adapt SourceCodes with PreventAssignSourcesToDefaultStockPlugin logic

maghamed committed:
Fixed Sample File for Inventory Import

Commit:
MSI: issue #290
- MEQP2 fixes;
- SourceImportExport integration tests fixes - part;
- Stock Source Import: modified sample file;

Commit:
Merge branch 'task-290-import-export-source-code' of https://github.com/magento-engcom/msi into task-290-import-export-source-code

Conflicts:
	app/code/Magento/InventoryImportExport/Files/Sample/stock_sources.csv

maghamed committed:
Fix conflicts with develop

maghamed committed:
Make Indexers to use SourceId instead of SourceCode

maghamed committed:
Make Indexers to use SourceId instead of SourceCode

maghamed committed:
Make Indexers to use SourceId instead of SourceCode

maghamed committed:
Remove Source Id from Admin UI

maghamed committed:
remove Source ID from API Interfaces

maghamed committed:
remove Source ID from API Interfaces

alexey-yakimovich committed:
Added setSourceId method to the Source model

alexey-yakimovich committed:
source_id field constant moved from 'Source' model to resource model

Commit:
Removed setSourceId method from source model; fixed integration test;

Commit:
Fixed integration test

Commit:
Fixed static test

Commit:
MSI: Task 290 import export source code

Commit:
MSI: Task 290 import export source code

Commit:
MSI: Task 290 import export source code

Commit:
MSI: Task 290 import export source code

Commit:
MSI: Task 290 import export source code

Commit:
MSI: Task 290 import export source code

Commit:
MSI: Task 290 import export source code

naydav commented:
Failed functional tests are related to
https://github.com/magento-engcom/msi/issues/333
